dtfs-central-api/src/services/portal/deal.service.ts

```diff
@@ -43,28 +43,38 @@ export class PortalDealService {
   }
 
   /**
-   * Add a "GEF deal cancellation pending" activity
-   * @param addGefDealCancellationPendingActivity
-   * @param addGefDealCancellationPendingActivity.dealId - the deal ID
-   * @param addGefDealCancellationPendingActivity.dealType - the deal type
-   * @param addGefDealCancellationPendingActivity.author - the activity's author
-   * @param addGefDealCancellationPendingActivity.auditDetails - the users audit details
+   * Adds a GEF deal cancelled activity to the portal activity repository.
+   * This function is executed for a future deal cancellation
+   * effective date.
+   *
+   * @param {Object} params - The parameters for adding the activity.
+   * @param {ObjectId | string} params.dealId - The ID of the deal.
+   * @param {DealType} params.dealType - The type of the deal.
+   * @param {ActivityAuthor} params.author - The author of the activity.
+   * @param {AuditDetails} params.auditDetails - The audit details for the activity.
+   * @param {Date} params.effectiveFrom - The date from which the cancellation is effective.
```

Comment:
I don't think we need the types for these params

Reply:
As discussed, to be convered in a separate PR for consistency.